variable/include/scipp/variable/bin_variable.tcc

```diff
@@ -108,11 +108,20 @@ template <class T> class BinVariableMaker : public BinVariableMakerCommon<T> {
 private:
   const VariableConstView
   bin_parent(const scipp::span<const VariableConstView> &parents) const {
-    if (parents.empty())
+    constexpr auto is_bins = [](auto &x) {
+      return x.dtype() == dtype<bucket<T>>;
+    };
+    const auto count = std::count_if(parents.begin(), parents.end(), is_bins);
+    if (count == 0)
       throw except::BinnedDataError("Bin cannot have zero parents");
-    return parents.front().dtype() == dtype<bucket<T>>
-               ? parents.front()
-               : bin_parent(parents.subspan(1));
+    if (!std::is_same_v<T, Variable> && count > 1)
+      throw except::BinnedDataError(
+          "Binary operations such as '+' with binned data are only supported "
+          "with dtype=VariableView, got dtype=DataArrayView or similar. See "
```

Comment:
👍  Good info. Very minor: could you pass actual received dtype as arg rather than hardcoding?